src/video/wayland/SDL_waylandevents.c

```diff
@@ -289,6 +289,24 @@ Wayland_SendWakeupEvent(_THIS, SDL_Window *window)
     WAYLAND_wl_display_flush(d->display);
 }
 
+static int
+dispatch_queued_events(SDL_VideoData *viddata)
+{
+    int ret;
+
+#ifdef HAVE_LIBDECOR_H
+    if (viddata->shell.libdecor) {
+        ret = libdecor_dispatch(viddata->shell.libdecor, 0);
+        if (ret < 0) {
+            return ret;
```

Comment:
Do we want to fail here and not call `WAYLAND_wl_display_dispatch_pending()` if libdecor has an issue?

Reply:
The assumption was that a dispatch error is something serious like a disconnection, but that should be caught by SDL as well when it tries to dispatch the queue immediately after this. This check can probably go.

Reply:
If not a check, probably a note for #6213 if we need to care about libdecor's involvement with reconnect support.